modin/numpy/arr.py

```diff
@@ -1347,6 +1347,60 @@ def __rmul__(
     ):
         return self.multiply(x2, out, where, casting, order, dtype, subok)
 
+    def dot(self, other, out=None):
+        other = try_convert_from_interoperable_type(other)
+        if numpy.isscalar(other):
+            # other is scalar -- result is an array
+            result = self._query_compiler.mul(other)
+        elif not isinstance(other, array):
+            raise TypeError(
+                f"Unsupported operand type(s): '{type(self)}' and '{type(other)}'"
+            )
+        elif self._ndim == 1 and other._ndim == 1:
+            # both 1D arrays -- result is a scalar
+            result = self._query_compiler.dot(
+                other._query_compiler, squeeze_self=True, squeeze_other=True
+            )
+            return result.to_numpy()[0, 0]
+        elif self._ndim == 2 and other._ndim == 2:
+            # both 2D arrays -- result is a 2D array
+            result = self._query_compiler.dot(other._query_compiler)
+        elif self._ndim == 1 and other._ndim == 2:
+            result = self._query_compiler.dot(other._query_compiler)
+        elif self._ndim == 2 and other._ndim == 1:
+            if other.shape[0] == 1:
+                result = self._query_compiler.dot(other._query_compiler.transpose())
+            else:
+                result = self._query_compiler.dot(other._query_compiler)
+        if out is not None:
+            out._query_compiler = result
+        return array(_query_compiler=result)
+
+    def _norm(self, ord=None, axis=None, keepdims=False):
+        check_kwargs(keepdims=keepdims)
+        if ord is not None and ord not in ("fro",):  # , numpy.inf, -numpy.inf, 0):
+            raise NotImplementedError("unsupported ord argument for norm:", ord)
+        apply_axis = axis or 0
```

Comment:
We need to account for negative axes as well.

Reply:
It seems like negative axes are supported (`axis=-1` is equivalent to `axis=1` for a 2D array, and `axis=-2` to `axis=0`). I'll add in a special case for this, though it seems like most of our other functions don't fully support this behavior.